source/scripts/script.js

```diff
@@ -19,54 +16,63 @@ async function init() {
         }
     });
 
+    // Add click event listener for search button
+    const searchButton = document.getElementById('search-button');
+    searchButton.addEventListener('click', async () => {
+        let searchSuccessful = await search();
+        if(searchSuccessful) {
+            createRecipeCards();
+        }
+    });
+
     // // Make the "Show more" button functional
     // bindShowMore();
-
 }
 
-
+// The search function, calls API function to fetch all recipes
+// Generates recipe cards by passing in values into RecipeData
 function search() {
-    // let searchQuery = document.getElementById('search-query').value;
-    // console.log(searchQuery);
-    // console.log(localStorage.getItem("dietaryRestrictions"));
-    //hideCategoryCards();
+    // get the search query
+    const searchQuery = document.getElementById('search-query').value;
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/cse110-fa21-group23/cse110-fa21-group23/pullRequest?prid=8408391)
```suggestion
    const searchQuery = document.getElementById("search-query").value;
```